feat: comprehensive Content AI documentation updates

ThisDavidRichard opened the pull request:
- Condensed AI Toolkit and AI Generation descriptions with custom icons
- Reorganized use cases page: removed search/filters, added contextual icons, reduced from 28 to 19 focused use cases
- Clarified custom LLM support and Tiptap AI server integration options
- Added deprecation notices to legacy extensions (AI Suggestion, AI Changes, AI Agent)
- Updated AI Toolkit onboarding flow with Business plan requirements
- Removed confusing UI elements and improved navigation consistency

ThisDavidRichard committed:
feat: comprehensive Content AI documentation updates

- Condensed AI Toolkit and AI Generation descriptions with custom icons
- Reorganized use cases page: removed search/filters, added contextual icons, reduced from 28 to 19 focused use cases
- Clarified custom LLM support and Tiptap AI server integration options
- Added deprecation notices to legacy extensions (AI Suggestion, AI Changes, AI Agent)
- Updated AI Toolkit onboarding flow with Business plan requirements
- Removed confusing UI elements and improved navigation consistency

ThisDavidRichard committed:
fix: correct import order in UseCasesGrid.tsx

ThisDavidRichard committed:
fix: correct import order for ESLint (relative before alias imports)

ThisDavidRichard committed:
fix: add required href property to all SidebarGroup items in legacy extension sidebars

svenadlung commented:
For the react installation instructions, installing `@tiptap/pm` is missing:

```
Install the core Tiptap packages and the [Vercel AI SDK](https://ai-sdk.dev/) for OpenAI:

npm install @tiptap/react @tiptap/starter-kit ai @ai-sdk/react @ai-sdk/openai
```

svenadlung committed:
fix: typo

ThisDavidRichard committed:
feat: enhance AI Toolkit documentation with improved layout and interactive demo

- Add caption feature to CodeDemo component with gradient background
- Integrate live demo iframe with caption on overview page
- Restructure overview page with Core Capabilities section (AI Agent Tools and Primitives)
- Add Guides and Examples section with CardGrid layout
- Create dedicated Install and Live Demo pages
- Update ProductCard to support custom secondary button text
- Add deprecated callouts to migration guide pages
- Improve PrevNextTiles alignment with body content
- Update sidebar navigation structure

ThisDavidRichard committed:
Merge branch 'content-ai-docs-updates' of https://github.com/ueberdosis/tiptap-docs into content-ai-docs-updates

ThisDavidRichard committed:
fix: prettier formatting for CodeDemo component

ThisDavidRichard committed:
Merge branch 'main' into content-ai-docs-updates

ThisDavidRichard committed:
feat: improve Content AI documentation and UI components

- Update metadata for better SEO across Content AI pages
- Add interactive demo caption with gradient background and refined typography
- Implement dynamic width for PrevNextTiles based on sidebar visibility
- Fix category mismatch and enhance descriptions for search optimization

ThisDavidRichard committed:
docs home adjustments

ThisDavidRichard committed:
Merge branch 'main' into content-ai-docs-updates

bdbch commented on src/components/PrevNextTiles.tsx:
```diff
@@ -5,6 +5,7 @@ import type { SidebarConfig, SidebarGroup, SidebarLink } from '@/types'
 type Props = {
   config?: SidebarConfig
   currentPath: string
+  hideSecondarySidebar?: boolean
```
Sounds weird because those tiles don't have anything to do with the secondary sidebar. I'd call it something like `isFullWidth`


```suggestion
  isFullWidth?: boolean
```

bdbch commented on src/components/CodeDemo.tsx:
```diff
@@ -5,25 +5,62 @@ import { DEMO_URL, PRO_DEMO_URL } from '@/utils/constants'
 
   const iframeProps = {
-    src: src ? src : (isPro ? PRO_DEMO_URL : DEMO_URL) + path,
+    src: src ? src : (isPro ? PRO_DEMO_URL : DEMO_URL) + (path || ''),
```
What's the reason having the path being an empty string? That would just load the root path of the demo page, not any demo?

bdbch commented on src/app/[...markdownPath]/page.tsx:
```diff
@@ -154,6 +154,7 @@ export default async function MarkdownPage({ params }: Props) {
           <PrevNextTiles
             config={sidebar.sidebarConfig}
             currentPath={`/${params.markdownPath.join('/')}`}
+            hideSecondarySidebar={!!pageMdx.frontmatter?.sidebars?.hideSecondary}
```
```suggestion
            isFullWidth={!!pageMdx.frontmatter?.sidebars?.hideSecondary}
```

bdbch commented on src/components/EnterpriseCallout.tsx:
```diff
@@ -5,38 +5,53 @@ import { Button } from './ui/Button'
+    { variant, inline = false, disableWaitlist = false, inverted = false, className, ...rest },
+    ref,
+  ) => {
     const title =
```
Not really related to the PR itself but this code looks crazy. I'd rather set up a dictionary object that hols which variant is what string and then do a `dict[variant]` here.

bdbch replied:
Oh seems like the comment was inserted at the wrong position. I'll add it at the correct one.

bdbch commented on src/components/PrevNextTiles.tsx:
```diff
@@ -70,7 +71,7 @@ function Card({ href, title, isNext }: { href: string; title: string; isNext?: b
 }
 
-export default function PrevNextTiles({ config, currentPath }: Props) {
+export default function PrevNextTiles({ config, currentPath, hideSecondarySidebar }: Props) {
```
```suggestion
export default function PrevNextTiles({ config, currentPath, isFullWidth }: Props) {
```

bdbch commented on src/components/PrevNextTiles.tsx:
```diff
@@ -87,7 +88,7 @@ export default function PrevNextTiles({ config, currentPath }: Props) {
 
   return (
-    <div className="mt-12">
+    <div className={`mt-12 ${hideSecondarySidebar ? 'max-w-full' : ''}`}>
```
```suggestion
    <div className={`mt-12 ${isFullWidth ? 'max-w-full' : ''}`}>
```

bdbch reviewed (changes requested):
Few minor things

ThisDavidRichard committed:
Update src/components/PrevNextTiles.tsx

Co-authored-by: bdbch <6538827+bdbch@users.noreply.github.com>

ThisDavidRichard committed:
Update src/app/[...markdownPath]/page.tsx

Co-authored-by: bdbch <6538827+bdbch@users.noreply.github.com>

bdbch commented on src/components/EnterpriseCallout.tsx:
```diff
@@ -5,38 +5,53 @@ import { Button } from './ui/Button'
+    ref,
+  ) => {
     const title =
       variant === 'migration'
```
Not really related to the PR itself but this code looks crazy. I'd rather set up a dictionary object that hols which variant is what string and then do a `dict[variant]` here.

ThisDavidRichard committed:
Update src/components/PrevNextTiles.tsx

Co-authored-by: bdbch <6538827+bdbch@users.noreply.github.com>

ThisDavidRichard committed:
Update src/components/PrevNextTiles.tsx

Co-authored-by: bdbch <6538827+bdbch@users.noreply.github.com>

bdbch committed:
Enforce required 'path' prop in CodeDemo component and update iframe source logic
